inst/include/sampler.h

```diff
@@ -157,8 +155,8 @@ namespace smc {
         const population<Space> & GetHistoryPopulationRefs(long n) const {return History[n].GetValues();}
         ///Returns the history flags
         historyflags GetHistoryFlags(long n) const {return History[n].GetFlags();}
-        ///Returns the history effective sample size.
-        long GetHistoryESS(long n) {return History[n].GetESS();}
+        /// Returns the Effective Sample Size of the specified particle generation.
+        long GetESS(long n) {return History[n].GetESS();}
```

Comment:
Sorry, why should this function return a long rather than a double? ESS is conveniently interpreted as an effective sample size, but its real valued rather than an integer... 

(I'd also make this function const for consistency with the other getters.)

Reply:
True. I think I will make it to return double. Also the const will be a great addition. To keep the programming style in line.